### PR TITLE
fix(hierarchy-list): add prop for hierarchy menus

### PR DESCRIPTION
**Pull request by JoelArmendariz**
Closes https://github.com/carbon-design-system/carbon-addons-iot-react/issues/1495

**Summary**

When an `OverflowMenu` is used in the `HierarchyList`, the menu is positioned incorrectly because it it implemented as a portal. As a result it is positioned absolute, relative to the body instead of the container we would expect.

Relevant issue in Monitor: https://github.ibm.com/wiotp/monitoring-dashboard/issues/1125

Relevant issue in Carbon: https://github.com/carbon-design-system/carbon/issues/4755

**Change List (commits, features, bugs, etc)**

Adding `data-floating-menu-container` and `style={{ position: 'relative' }}` to a wrapper div forces the menu to be placed back into the DOM where we would expect it, and to be positioned relative to the `HierarchyList` container itself, rather than the body

**Acceptance Test (how to verify the PR)**

Go to the `With OverflowMenu` story  and verify that the position stays with the menu icon through the `scrollIntoView` and as you scroll.

### CODE REVIEW

File: src/components/List/HierarchyList/HierarchyList.jsx [hunk 1]
```diff
@@ -264,39 +264,41 @@ const HierarchyList = ({
...
-      }
-      items={pageSize != null ? itemsToShow : filteredItems}
-      expandedIds={expandedIds}
-      toggleExpansion={id => {
-        if (expandedIds.filter(rowId => rowId === id).length > 0) {
-          // remove id from array
-          setExpandedIds(expandedIds.filter(rowId => rowId !== id));
-        } else {
-          setExpandedIds(expandedIds.concat([id]));
+    <div data-floating-menu-container style={{ position: 'relative' }}>
```
Comment (tay1orjones) [inline src/components/List/HierarchyList/HierarchyList.jsx:267]: This style should be in the .scss file and applied via a class. Do we need the data attribute for something?

Comment (sstone2423) [inline src/components/List/HierarchyList/HierarchyList.jsx:267]: Make sure you add the comment and carbon issue link here so others know why its here

Comment (JoelArmendariz) [reply src/components/List/HierarchyList/HierarchyList.jsx:267]: I have the addons issue linked in the PR description already. I will add the carbon issue to the description.

Comment (JoelArmendariz) [reply src/components/List/HierarchyList/HierarchyList.jsx:267]: I will extract to scss.

The data attribute forces the Menu to be in the DOM as a child of what is rendering it. Currently the implementation of this in Carbon has it as a portal (https://reactjs.org/docs/portals.html) which places it outside of our tree of components. I have updated the description of this PR to have the relevant Carbon issue that had reference to this fix.

Comment (JoelArmendariz) [reply src/components/List/HierarchyList/HierarchyList.jsx:267]: This doesn't seem to be an issue for `SimpleList`. https://carbon-addons-iot-react.com/?path=/story/watson-iot-experimental-simplelist--list-with-overflow-menu

Comment (sstone2423) [reply src/components/List/HierarchyList/HierarchyList.jsx:267]: At the top of the PR description, set this PR to close that addons issue.

I meant to add a code comment above that `data-floating-menu-container` so developers will understand why its there

Comment (JoelArmendariz) [reply src/components/List/HierarchyList/HierarchyList.jsx]: Yes. I added it. Sorry, I miss-read.

**Comment by JoelArmendariz** [comment]
@tay1orjones I have moved the data attribute and position fixed to live on the list item, which fixes the scrolling issue. Thanks for the suggestion!

**Pull request merged**